### ISSUE TITLE
Fonts activated by FontBase are not found

### ISSUE THREAD

**Issue by ghost** [moi15moi/FindSystemFontsFilename#14]
Reopening https://github.com/moi15moi/FontCollector/issues/40 here

> (Windows 11)
> 
> The current font enumeration method does not find fonts activated by [FontBase](https://fontba.se/); even though they should be detected as system fonts considering every app with a font picker can find them.
> 
> Potentially could also be causing issues with other apps that utilize the same method as FontBase to activate their fonts.
> 
> I've confirmed with other users that they also experience the same issue. One user have testified that FontCollector successfully finds their FontBase fonts though but we couldn't identify the cause—so there might be a configuration change necessary for it to work.

**Comment by moi15moi** [comment]
Thanks for opening an issue.
I will need your help to debug this problem because I am not able to reproduce the problem.

On the issue you opened in FontCollector, you said:
> The fonts can be found at the registry key: ``HKEY_CURRENT_USER\Software\Microsoft\Windows NT\CurrentVersion\Fonts``

There are many reasons why we can't use this registry key to get the fonts. One of them is that it doesn't collect system fonts like ``Marlett``.

I have some questions for you:
1. Do you see the activated font in the registry?
2. Have you rebooted? In some very specific moments, DirectWrite uses cache and doesn't refresh it, which can miss some activated fonts. This is a bug that I can do nothing to correct it.
3. What is the file path of the activated font by FontBase? The font should be in ``%USERPROFILE%\FontBase``.

**Comment by ghost** [comment]
> Do you see the activated font in the registry?

Yes

> Have you rebooted? In some very specific moments, DirectWrite uses cache and doesn't refresh it, which can miss some activated fonts. This is a bug that I can do nothing to correct it.

I haven't. I can try; am I supposed to reboot _after_ enabling fonts?
Maybe there's a different _working_ way to find these fonts? I assume there is, because other software with font pickers e.g. Aegisub are able to display them perfectly fine.

> What is the file path of the activated font by FontBase? The font should be in %USERPROFILE%\FontBase.

You're correct, it's in that directory

```
C:\Users\yuv420p10le\FontBase\GANDHISANS-BOLD.OTF
```

**Comment by moi15moi** [comment]
> I haven't. I can try; am I supposed to reboot _after_ enabling fonts?

Exactly. Reboot and reactivate it.

> Maybe there's a different _working_ way to find these fonts? I assume there is, because other software with font pickers e.g. Aegisub are able to display them perfectly fine.

Aegisub and any other tools use GDI and/or DirectWrite which is also what this tool use. There is no reason that this tool doesn't find it.

Could you tell me which version of FindSystemFontsFilename you have?
You can get it with ``pip show FindSystemFontsFilename``. If it is below 0.2.0, update the package with ``pip install FindSystemFontsFilename -U`` and run this
```py
from find_system_fonts_filename import get_system_fonts_filename

fonts = get_system_fonts_filename()

for font in fonts:
    if font.lower() == r'C:\Users\yuv420p10le\FontBase\GANDHISANS-BOLD.OTF'.lower():
        print(f"Found at {font}")
```

**Comment by ghost** [comment]
Rebooted, activated fonts. Ensured I'm on 2.0.

The snippet does not reach the print statement either when the font is activated or not.

**Comment by moi15moi** [comment]
Finally, I am able to reproduce the issue. I am using a VM that is on 10.0.22631.2861.
This issue seems to only happen on Windows 11 which is a bit strange since there isn't any big update that only impacts Windows 11 (information from [here](https://learn.microsoft.com/en-us/windows/win32/directwrite/what-s-new-in-directwrite-for-windows-8-consumer-preview)).

> One user have testified that FontCollector successfully finds their FontBase fonts though but we couldn't identify the cause

~~Does this user is on Windows 10? For me, it works fine on Windows 10.~~
The user is on Windows 11. Specifically 10.0.22631.3235

**Comment by moi15moi** [comment]
@yuv420p10le Could you confirm me that #15 fix your issue?

**Comment by ghost** [comment]
The issue is resolved on fontcollector v3.0.0rc2 with #15 applied so I'm gonna say yes

**Comment by moi15moi** [comment]
That doesn't make any sense.
Anyways, since now it also collect the fonts added with AddFontRessource, everything should be alright.
